tests/functional_tests/ffront_tests/test_program.py

```diff
@@ -278,3 +278,48 @@ def empty_domain_program(
         match=(r"Dimensions in out field and field domain are not equivalent"),
     ):
         empty_domain_program(a, out_field, offset_provider={})
+
+
+def test_input_kwargs(fieldview_backend):
+    size = 10
+    input_1 = np_as_located_field(IDim, JDim)(np.ones((size, size)))
+    input_2 = np_as_located_field(IDim, JDim)(np.ones((size, size)) * 2)
+    input_3 = np_as_located_field(IDim, JDim)(np.ones((size, size)) * 3)
+    out = np_as_located_field(IDim, JDim)(np.zeros((size, size)))
+
+    expected_fieldop = np.asarray(input_3) * np.asarray(input_1) - np.asarray(input_2)
+    expected_program = np.asarray(input_3) * np.asarray(input_2) - np.asarray(input_1)
+
+    @field_operator(backend=fieldview_backend)
+    def fieldop_input_kwargs(
+        a: Field[[IDim, JDim], float64],
+        b: Field[[IDim, JDim], float64],
+        c: Field[[IDim, JDim], float64],
+    ) -> Field[[IDim, JDim], float64]:
+        return c * a - b
+
+    fieldop_input_kwargs(input_1, b=input_2, c=input_3, out=out, offset_provider={})
+    assert np.allclose(expected_fieldop, out)
+
+    @program
+    def program_input_kwargs(
+        a: Field[[IDim, JDim], float64],
+        b: Field[[IDim, JDim], float64],
+        c: Field[[IDim, JDim], float64],
+        out: Field[[IDim, JDim], float64],
+    ):
+        fieldop_input_kwargs(c, a, b, out=out)
+
+    out = np_as_located_field(IDim, JDim)(np.zeros((size, size)))
+
+    program_input_kwargs(input_1, b=input_2, c=input_3, out=out, offset_provider={})
+    assert np.allclose(expected_program, out)
+    out = np_as_located_field(IDim, JDim)(np.zeros((size, size)))
+
+    program_input_kwargs(a=input_1, b=input_2, c=input_3, out=out, offset_provider={})
+    assert np.allclose(expected_program, out)
+
+    with pytest.raises(GTTypeError) as exc_info_1:
```

Comment:
Since this is not a multi-level-exception (or however they are I called, I always forget, I mean the `raise ... from e` where one is interested in the message in `e`), you can use the regular pytest pattern. Here is the excerpt from the [docs](https://docs.pytest.org/en/7.1.x/how-to/assert.html):
```python
def test_match():
    with pytest.raises(ValueError, match=r".* 123 .*"):
        myfunc()
```